crates/router/src/connector/authorizedotnet.rs

```diff
@@ -407,7 +425,14 @@ impl ConnectorIntegration<api::Void, types::PaymentsCancelData, types::PaymentsR
         &self,
         req: &types::PaymentsCancelRouterData,
     ) -> CustomResult<Option<types::RequestBody>, errors::ConnectorError> {
-        let connector_req = authorizedotnet::CancelOrCaptureTransactionRequest::try_from(req)?;
+        let connector_router_data = authorizedotnet::AuthorizedotnetRouterData::try_from((
+            &self.get_currency_unit(),
+            req.request.currency.unwrap(),
```

Comment:
amount is not required for Authorizedotnet cancel flow  . You can revert changes for void flow . And also unwrap shouldn't be used as application goes to panic state instead throw appropriate error.

Reply:
CancelOrCaptureTransactionRequest is used for cancel request and capture request both, if I revert for cancel then I'll have to do the same for capture request